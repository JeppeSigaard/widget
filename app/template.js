const template = (html, initialState) => `
  <!DOCTYPE html>
  <html lang="en">
  <head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <title>Weather widget</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
  </head>
  <body>
  <div id='entry'>${html}</div>
  <script>window.initialState=${JSON.stringify(initialState)}</script>
  <script src='/app.js'></script>
  </body>
  </html>
`;

export default template;
