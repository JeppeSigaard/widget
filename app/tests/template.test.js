import template from "../template";

describe("Template file", () => {
  it("returns a template", () => {
    expect(template("mockhtml", { sum: "data" })).toMatchSnapshot();
  });
});
