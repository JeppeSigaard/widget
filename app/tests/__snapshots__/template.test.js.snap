// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template file returns a template 1`] = `
"
  <!DOCTYPE html>
  <html lang=\\"en\\">
  <head>
  <meta http-equiv=\\"content-type\\" content=\\"text/html; charset=utf-8\\">
  <title>Weather widget</title>
  <link rel=\\"stylesheet\\" href=\\"https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css\\" integrity=\\"sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u\\" crossorigin=\\"anonymous\\">
  </head>
  <body>
  <div id='entry'>mockhtml</div>
  <script>window.initialState={\\"sum\\":\\"data\\"}</script>
  <script src='/app.js'></script>
  </body>
  </html>
"
`;
